src/GNDStk/BodyText/src/sync.hpp

```diff
@@ -0,0 +1,46 @@
+
+// -----------------------------------------------------------------------------
+// pullFromDerived(content)
+// Make this BodyText's length, start, and valueType parameters be consistent
+// with any or all such parameters that exist in the given content. Remember
+// that BodyText (this class) is a base of Component, which is a base of some
+// other class. The final derived class is what contains the "content" struct.
+// -----------------------------------------------------------------------------
+
+template<class CONTENT>
+void pullFromDerived(const CONTENT &content)
+{
+   // length?
+   if constexpr (detail::has_length<CONTENT>::value)
+      length(content.length);
+
+   // start?
+   if constexpr (detail::has_start<CONTENT>::value)
+      start(content.start);
+
+   // valueType?
+   if constexpr (detail::has_valueType<CONTENT>::value)
+      valueType(content.valueType);
+}
+
+
+// -----------------------------------------------------------------------------
+// pushToDerived(content)
+// Basically, the reverse of the above.
+// -----------------------------------------------------------------------------
+
+template<class CONTENT>
+void pushToDerived(CONTENT &content) const
```

Comment:
See above, I don't think the SFINAE works properly in these two functions due to the derived class being incomplete (same issue as the has_construct we have had to deal with).